implementations/rust/ockam/ockam_command/src/node/show.rs

```diff
@@ -52,10 +52,51 @@ async fn run_impl(
     ctx: Context,
     (opts, cmd): (CommandGlobalOpts, ShowCommand),
 ) -> miette::Result<()> {
-    let node_name = get_node_name(&opts.state, &cmd.node_name);
+    if let Some(node_name) = cmd.node_name {
+        show_node(&opts, &ctx, &node_name).await?;
+    } else if !opts.terminal.can_ask_for_user_input() {
+        show_node(&opts, &ctx, &get_default_node_name(&opts.state)).await?;
+    } else {
+        let node_names = opts.state.nodes.list_items_names()?;
+        if node_names.is_empty() {
+            opts.terminal
+                .stdout()
+                .plain("There are no nodes to show")
+                .write_line()?;
+            return Ok(());
+        }
+
```

Comment:
If the `node_names` list only contains one item, it should be shown right away without interaction. This would mean that you have three cases: the user has selected either 0, 1 or >1. So maybe something like this would be more explicit:

```rust
match node_names.len() {
    0 => {
        opts.terminal
            .stdout()
            .plain("There are no nodes to show")
            .write_line()?;
    }
    1 => {
        show_node(&opts, &ctx, &node_names[0]).await?;
    }
    _ => {
        let selected_node_names = opts.terminal.select_multiple(
            "Select one or more nodes that you want to show".to_string(),
            node_names,
        );
        ...
    }
}
```